drivers/SAI.cpp

```diff
@@ -50,6 +53,10 @@ bool SAI::xfer(uint32_t *value) {
     return ret;
 }
 
+sai_result_t SAI::status(void) {
```

Comment:
As per the comment there : https://github.com/ARMmbed/Handbook/pull/451#discussion_r182133333
I don't know if "status" is a correct wording here as the `status` would be `uninitialized` or `not ready` rather than a `Invalid param` or `config mismatch` that are the **reasons** why the `status` is not OK.

Reply:
It also exposes a type directly from the HAL.
@kegilbert @c1728p9 Is this commonly accepted ?

Reply:
I believe yes, redefining them is not necessary (creating a wrapper in drivers) , as HAL defines them already and should not change.

Reply:
Would you prefer `result` as the wording rather than `status` @ithinuel ?

Reply:
@kegilbert No as here we're reporting the status of the instance (`uninitialised with error ...`).
@0xc0170 confirmed that it's fine to expose HAL's types.
And even if the type ever change C++ consider enum as proper types (which is not the case in C) and would fail to compile .